javatests/arcs/core/storage/StorageProxyTest.kt

```diff
@@ -549,32 +623,60 @@ class StorageProxyTest {
     }
 
     @Test
-    fun closeStorageProxy_closesStoreEndpoint() = runBlocking {
+    fun closeStorageProxy_closesStoreEndpoint() = runTest {
         val proxy = StorageProxy(mockStorageEndpointProvider, mockCrdtModel, scheduler)
         proxy.close()
         assertThat(fakeStoreEndpoint.closed).isTrue()
         assertThat(proxy.getStateForTesting()).isEqualTo(ProxyState.CLOSED)
     }
 
     // Convenience wrapper for destructuring.
-    private data class ActionMocks (
+    private data class ActionMocks(
         val onReady: () -> Unit = mock(),
         val onUpdate: (String) -> Unit = mock(),
         val onDesync: () -> Unit = mock(),
-        val onResync: () -> Unit = mock()
+        val onResync: () -> Unit = mock(),
+        val channels: ActionChannels
```

Comment:
Why do we need the channels here? They're both invoked and checked from test code (i.e. IIUC they don't interact with the StorageProxy channels at all), so how do they differ from the existing mock-based checks?